DigitalLearningSolutions.Data/Services/CourseService.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models.Courses;
+
+    public interface ICourseService
+    {
+        IEnumerable<CourseStatistics> GetTopCourseStatisticsAtCentreForCategoryId(int centreId, int categoryId);
+    }
+
+    public class CourseService : ICourseService
+    {
+        private readonly ICourseDataService courseDataService;
+
+        public CourseService(ICourseDataService courseDataService)
+        {
+            this.courseDataService = courseDataService;
+        }
+
+        public IEnumerable<CourseStatistics> GetTopCourseStatisticsAtCentreForCategoryId(int centreId, int categoryId)
+        {
+            var allCourses = courseDataService.GetCourseStatisticsAtCentreForCategoryId(centreId, categoryId);
+            return allCourses.Where(c => c.Active).OrderByDescending(c => c.DelegateCount);

```

Comment:
Note that we are ordering here by the number of delegates. The old site appears to order by the number In Progress, so there may be slight discrepancies as to what appears in what order in the top courses pages.

Reply:
It was decided to order by the number In Progress during Show & Tell on 02/07/2021. Ticket description changed accordingly.